### PR TITLE
All needed changes to productize 2.7

### PR DESCRIPTION
**Pull request by miguelsorianod**
I Open this PR to prepare and review all the needed changes to productize 2.7 from the master branch.

### CODE REVIEW

**Comment by miguelsorianod** [comment]
Additional notes:

* Values of images still not decided what those will be due to maybe there's a change in our url schemes. Also we should decide where to put brew urls and where to put "definitive" urls

File: deploy/olm-catalog/3scale-operator/3scale-operator.package.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+packageName: 3scale-operator
+channels:
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply deploy/olm-catalog/3scale-operator/3scale-operator.package.yaml]: Seems a good idea, but there is some case that concerns me a little bit.

If we update alpha channel CurrentCSV to 0.4.0 that would mean that possible users that have already installed 3scale 2.6 (and thus are already in alpha channel)will be updated to 3scale 2.7 (if they have automatic updates enabled). Is that what we want?

File: deploy/olm-catalog/3scale-operator/3scale-operator.v0.4.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+metadata:
+  name: 3scale-operator.v0.4.0
+  namespace: placeholder
+  annotations:
+    repository: https://github.com/3scale/3scale-operator
+    capabilities: Basic Install
+    categories: "Integration & Delivery"
+    certified: "false"
+    description: 3scale Operator to provision 3scale and publish/manage API
+    containerImage: proxy.engineering.redhat.com/rh-osbs/3scale-amp2/3scale-rhel7-operator:3scale2.7
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply deploy/olm-catalog/3scale-operator/3scale-operator.v0.4.0.clusterserviceversion.yaml]: :+1: I'll add it

File: deploy/olm-catalog/backup-3scale-operatorsource.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apiVersion: operators.coreos.com/v1
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply deploy/olm-catalog/backup-3scale-operatorsource.yaml]: :+1:

File: .circleci/config.yml [hunk 4]
```diff
@@ -299,6 +311,7 @@ jobs:
       - checkout
       - install-openshift
       - create-secrets
+      - create-redhat-registry-io-secret
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply .circleci/config.yml]: I thought about that. It should only be executed on master branch due to the 'filters' attribute. The reason I did not remove it is to try to reduce the differences between the branches, but if you prefer us to remove it no problem.

File: deploy/olm-catalog/3scale-operator/3scale-operator.package.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (eguzki) [reply deploy/olm-catalog/3scale-operator/3scale-operator.package.yaml]: What is the meaning of alpha channel? What, as a customer, do you expect from it? I would expect to receive all update. And you?

[the remark this answers is not in the document]
Comment (miguelsorianod) [reply deploy/olm-catalog/3scale-operator/3scale-operator.package.yaml]: Yes, in the new concept of "alpha" I would expect that.
The concern is with the current users of that channel, where we are really offering 2.6 release (in a stable way)

**Comment by miguelsorianod** [comment]
Updated PR with rewriting of commit history

File: deploy/olm-catalog/3scale-operator/3scale-operator.v0.4.0.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+metadata:
+  name: 3scale-operator.v0.4.0
+  namespace: placeholder
+  annotations:
+    repository: https://github.com/3scale/3scale-operator
+    capabilities: Basic Install
+    categories: "Integration & Delivery"
+    certified: "false"
+    description: 3scale Operator to provision 3scale and publish/manage API
+    containerImage: registry.redhat.io/3scale-amp2/3scale-rhel7-operator:3scale2.7
```
Comment (miguelsorianod) [inline deploy/olm-catalog/3scale-operator/3scale-operator.v0.4.0.clusterserviceversion.yaml]: Should we set the -{RELEASE} suffix? if so, what number? lets start with -1 ?

File: .circleci/config.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (eguzki) [reply .circleci/config.yml]: this new line does not have any effect... but leave it

File: deploy/olm-catalog/3scale-operator/3scale-operator.v0.4.0.clusterserviceversion.yaml [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+
+    All bugs, tasks or enhancements are tracked as [GitHub issues](https://github.com/3scale/3scale-operator/issues).
+
+    ### License
+    3scale Operator is licensed under the [Apache 2.0 license](https://github.com/3scale/3scale-operator/blob/master/LICENSE)
+
+  keywords:
+    - 3scale
+    - API
+  displayName: 3scale Operator
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply deploy/olm-catalog/3scale-operator/3scale-operator.v0.4.0.clusterserviceversion.yaml]: :+1:

File: deploy/olm-catalog/3scale-operator/3scale-operator.v0.4.0.clusterserviceversion.yaml [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+  maturity: alpha
+  provider:
+    name: Red Hat
+  links:
+    - name: GitHub
+      url: https://github.com/3scale/3scale-operator
+    - name: Documentation
+      url: https://github.com/3scale/3scale-operator/blob/v0.4.0/doc/user-guide.md
+  maintainers:
+    - email: openshift-operators@redhat.com
```
[the remark this answers is not in the document]
Comment (miguelsorianod) [reply deploy/olm-catalog/3scale-operator/3scale-operator.v0.4.0.clusterserviceversion.yaml]: We'll decide what the email has to be in a future PR

**Pull request merged**